streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilder.java

```diff
@@ -2245,4 +2272,22 @@ public boolean hasNamedTopology() {
     public synchronized Map<String, StoreFactory> stateStores() {
         return stateFactories;
     }
+
+    public <KIn, VIn, VOut> WrappedFixedKeyProcessorSupplier<KIn, VIn, VOut> wrapFixedKeyProcessorSupplier(
+        final String name,
+        final FixedKeyProcessorSupplier<KIn, VIn,  VOut> processorSupplier
+    ) {
+        return ProcessorWrapper.asWrappedFixedKey(
+            processorWrapper.wrapFixedKeyProcessorSupplier(name, processorSupplier)
```

Comment:
Just checking if this is correct? `wrapFixedKeyProcessorSupplier` func itself calls `return ProcessorWrapper.asWrappedFixedKey(processorSupplier);` already, and hence seems a circular calling?

Reply:
Ah, yes, it's correct but I see how it's confusing -- basically the non-static `#wrapFixedKeyProcessorSupplier` is what does the actual wrapping via the configured ProcessorWrapper instance, whereas the static `ProcessorWrapper#asWrapped` method is what converts a regular ProcessorSupplier into the `WrappedProcessorSupplier` subclass that we use as a marker interface to distinguish between processors that are and aren't wrapped yet.

I will add a comment to the code to clarify for future readers, but perhaps it's also worth choosing a better name for the static `#asWrapped` utility methods. Maybe `#toWrappedProcessorSupplier` or something like that?

@guozhangwang @agavra thoughts?

(btw I am going to merge the PR as-is since we can address this in a followup PR, I'll make sure to add the code comments and/or change the name in another PR)

Reply:
I see, yeah I think having a different name for the static utility function would be better.